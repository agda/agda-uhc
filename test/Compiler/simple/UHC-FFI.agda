-- Checks that UHC FFI calls using non-magic datatypes work
module UHC-FFI where

data Nat : Set where
  Zero : Nat
  Suc : Nat -> Nat
{-# COMPILE UHC Nat = data UHC.Agda.Builtins.Nat (Zero | Suc) #-}

{-# FOREIGN UHC __IMPORT__ Data.Char #-}

open import Common.IO
open import Common.Unit
open import Common.String
open import Common.Char


postulate toLower : Char -> Char
{-# COMPILE UHC toLower = Data.Char.toLower #-}

main : IO Unit
main = putStr (charToStr (toLower 'A'))

