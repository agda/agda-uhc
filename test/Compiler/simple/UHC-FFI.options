TestOptions {forCompilers = [(UHC, CompilerOptions {extraAgdaArgs = []})], runtimeOptions = [], executeProg = True}
