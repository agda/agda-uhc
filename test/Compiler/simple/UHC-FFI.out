EXECUTED_PROGRAM

ret > ExitSuccess
out > a
